CenterForOpenScience/ember-osf-web
# [ENG-512] Dialog box

## Purpose
Add a reusable dialog box component that we can use for all our dialog boxing needs. Should be styled consistently with new page designs, be accessible and keyboard navigable.


## Summary of Changes
Add an `osf-dialog` component and a handbook page documenting it.


## Side Effects
Added a wormhole outlet in the application template, where dialogs will be rendered with ember-wormhole.

Added styles to `[inert]` to blur and gray out inert elements.


## Feature Flags
n/a


## QA Notes
You can check out the dialog on its handbook page:
https://centerforopenscience.github.io/ember-osf-web/handbook/docs/components/osf-dialog
(It may take some time for the handbook to deploy after this PR is merged).

Important behavior:
- Looks like the mockups (with an exception: footer buttons align to the right)
- When a dialog is open, the rest of the page is clearly inert/inaccessible
- The dialog should grow with its content, but never get larger than the browser viewport
- Keyboard navigation should be predictable:
  - Press `tab` to focus a button and press `enter` to open the dialog
  - Pressing `tab` with the dialog open should move focus within the dialog, but not let you focus anything else on the page
  - Pressing `esc` (while the focus is inside the dialog) or hitting the dialog's close button should close the dialog and return focus to where it was before
  - Note: there are some weirdnesses with tab navigation in firefox and safari, depending on your OS settings, and there's a known bug (with safari) in the library we're using to make the rest of the page inert -- don't worry too much about `tab` behavior as long as it works on chrome.



## Ticket


https://openscience.atlassian.net/browse/ENG-512

# Reviewer Checklist





[EMB-123]: https://openscience.atlassian.net/browse/EMB-123
